### PR TITLE
Web ide/aayushi

### PR DESCRIPTION
**Pull request by aayushitiwari03**
WebIde

### CODE REVIEW

File: app/src/main/java/org/merakilearn/datasource/network/model/UploadCredentials.kt [hunk 1]
```diff
@@ -12,25 +12,25 @@ data class UploadCredentials(
...
-    val Body: String?,
+    val Body : String?,
 
     @Json(name = "Credentials")
-    val Credentials: Credentials,
+    val Credentials : Credentials,
 
     @Json(name = "project_id")
-    val projectId: String
+    val project_id : String
```
Comment (damanpreet97) [inline app/src/main/java/org/merakilearn/datasource/network/model/UploadCredentials.kt]: this variable should be projectId and not in snake case, to maintain the convention

File: app/src/main/res/layout/dialog_create.xml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+                <com.google.android.material.textfield.TextInputEditText
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:hint="@string/enter_project_name"
+                    android:inputType="textNoSuggestions" />
+
+            </com.google.android.material.textfield.TextInputLayout>
+
+            <!--            <com.google.android.material.textfield.TextInputLayout-->
```
Comment (damanpreet97) [inline app/src/main/res/layout/dialog_create.xml:40]: pls remove commented code

Comment (surajsahani) [reply app/src/main/res/layout/dialog_create.xml:40]: @aayushitiwari03

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -11,8 +11,12 @@
     <uses-permission android:name="android.permission.MICROPHONE" />
     <uses-permission android:name="android.permission.CAMERA" />
     <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
+    <uses-permission android:name="android.permission.READ_CONTACTS" />
+    <uses-permission android:name="android.permission.WRITE_CONTACTS" />
+
 
     <application
+
```
Comment (surajsahani) [inline app/src/main/AndroidManifest.xml:19]: Remove extra space @aayushitiwari03

File: app/src/main/java/org/merakilearn/FileChooserActivity.kt [hunk 4]
```diff
@@ -96,7 +105,51 @@ class FileChooserActivity : AppCompatActivity() {
                 }
             }
         }
+    }
 
+    private fun checkAndRequestPermissions() {
```
Comment (surajsahani) [inline app/src/main/java/org/merakilearn/FileChooserActivity.kt:110]: Where we are using this permission @aayushitiwari03

**Review by surajsahani — CHANGES_REQUESTED** [review]
Please address the changes in comment

File: app/src/main/res/layout/dialog_create.xml (hunk 2 above)
Comment (aayushitiwari03) [reply app/src/main/res/layout/dialog_create.xml:40]: I can't remove any commented code of WebIDE as of now, in order to compare it with the original Hyper.

File: app/src/main/java/org/merakilearn/FileChooserActivity.kt (hunk 4 above)
Comment (aayushitiwari03) [reply app/src/main/java/org/merakilearn/FileChooserActivity.kt:110]: @surajsahani We are using this function inside on create method and this is important.

**Pull request merged**